concourse/pipelines/deploy.yml

```diff
@@ -15,49 +15,49 @@ resources:
     icon: github
     name: govuk-infrastructure
     source:
-      branch: main
+      branch: ((govuk-infrastructure-branch))
       uri: https://github.com/alphagov/govuk-infrastructure
     type: git
 
   - <<: *git-repo
     name: frontend
     source:
-      branch: master
+      branch: ((frontend-branch))
       uri: https://github.com/alphagov/frontend
       tag_filter: release_*
 
   - <<: *git-repo
     name: publisher
     source:
-      branch: master
+      branch: ((publisher-branch))
       uri: https://github.com/alphagov/publisher
       tag_filter: release_*
 
   - <<: *git-repo
     name: publishing-api
     source:
-      branch: master
+      branch: ((publishing-api-branch))
       uri: https://github.com/alphagov/publishing-api
       tag_filter: release_*
 
   - <<: *git-repo
     name: content-store
     source:
-      branch: master
+      branch: ((content-store-branch))
       uri: https://github.com/alphagov/content-store
       tag_filter: release_*
 
   - <<: *git-repo
     name: router
     source:
-      branch: master
+      branch: ((router-branch))
       uri: https://github.com/alphagov/router
       tag_filter: release_*
 
   - <<: *git-repo
     name: router-api
     source:
-      branch: master
+      branch: ((router-api-branch))
```

Comment:
One small danger to this approach is that it could require a `concourse/settings/production.yml` file, which can be modified to deploy a non-`master` branch to production. Perhaps a separate resource for integration might be in order, particularly as the `tag_filter` here is currently incompatible with branch deployments.

Reply:
Ah, good point about the `tag_filter`. Perhaps we need to pull that into the settings too...

Something like:

`test.yml`
```
---
git:
  router-api:
    branch: master
    tag_filter: null
```

`production.yml`
```
---
git:
  router-api:
    branch: master
    tag_filter: release_*
```

Perhaps we could write a test to enforce that staging / production only had master / main branches.

Reply:
(I think it's possible to do `((git.router-api.branch))` and `((git.router-api.tag_filter))` in concourse interpolations, but I haven't checked)

Reply:
I think we'll need to bite the bullet and build container images in the pipeline to be able to do branch deployments at all, come to think of it.

Reply:
That looks reasonable 👍